### PR TITLE
Remove hardcoded `top_n` value in `clustfcc`

### PR DESCRIPTION
**Pull request by rvhonorato**
# Pull Request template

You are about to submit a new Pull Request. Before continuing make sure
you read the [contributing guidelines](CONTRIBUTING.md) and you comply
with the following criteria:

- [X] Your code is well documented. Functions and classes have proper docstrings
  as explained in contributing guidelines. You wrote explanatory comments for
  those tricky parts.
- [ ] You wrote tests for the new code.
- [X] `tox` tests pass. Run `tox` command inside the repository folder.
- [X] `-test.cfg` examples execute without errors. Inside `examples/` run
  `python run_tests.py -b`
- [X] You have stick to Python. Talk with us before if you really need to add
  other programming languages to HADDOCK3
- [X] code follows our coding style
- [X] You avoided structuring the code into classes as much as possible, using
  small functions instead. But, you can use classes if there's a purpose.
- [X] PR does not add any *install dependencies* unless permission was granted
  by the HADDOCK team
- [X] PR does not break licensing
- [ ] You get extra bonus if you write tests for already existing code :godmode:

* * *

### CODE REVIEW

**Review by joaomcteixeira — COMMENTED** [review]
Before `top_n=4` always and now `threshold` is defined by the `for` loop in line 107. From what I understand `threshold` will be the first value for which clusters are identified. Is this what is intended (honest question)?

File: src/haddock/modules/analysis/clustfcc/__init__.py [hunk 1]
```diff
@@ -145,13 +145,12 @@ def _run(self):
...
             # Rank the clusters
-            #  they are sorted by the top4 models in each cluster
-            top_n = 4
+            #  they are sorted by the topX (threshold) models in each cluster
             score_dic = {}
             for clt_id in clt_dic:
                 score_l = [p.score for p in clt_dic[clt_id]]
                 score_l.sort()
-                top4_score = sum(score_l[:top_n]) / float(top_n)
+                top4_score = sum(score_l[:threshold]) / float(threshold)
```
Comment (amjjbonvin) [inline src/haddock/modules/analysis/clustfcc/__init__.py:153]: Is there somewhere a check that the threshold should be >= 1?

**Review by amjjbonvin — APPROVED** [review]

**Comment by amjjbonvin** [comment]
> 

Yes this is the intended behaviour.

File: src/haddock/modules/analysis/clustfcc/__init__.py (hunk 1 above)
Comment (rvhonorato) [reply src/haddock/modules/analysis/clustfcc/__init__.py:153]: just added!

Comment (joaomcteixeira) [reply src/haddock/modules/analysis/clustfcc/__init__.py:153]: Is defined by the for loop and the `range`

**Comment by joaomcteixeira** [comment]
:+1:

**Pull request merged**